src/rt/cover.c

```diff
@@ -1301,57 +1308,107 @@ static void cover_print_code_line(FILE *f, loc_t loc, cover_line_t *line)
    fprintf(f, "</code>");
 }
 
-static void cover_print_bins(FILE *f, cover_pair_t *pair)
+static void cover_print_get_exclude_button(FILE *f, cover_tag_t *tag,
+                                           uint32_t flag, bool add_td)
+{
+
+   const char *bin_name = "";
+   if (flag)
+      for (int i = 0; i < ARRAY_LEN(bin_map); i++)
+         if (bin_map[i].flag == flag)
+            bin_name = bin_map[i].name;
+
+   if (add_td)
+      fprintf(f, "<td>");
+
+   fprintf(f, "<button onclick=\"GetExclude('exclude %s %s')\" %s>"
+              "Copy %sto Clipoard</button>", istr(tag->hier), bin_name,
+           (tag->kind == TAG_STMT) ? "style=\"float: right;\"" : "",
+           (tag->kind == TAG_STMT) ? "Exclude Command " : "");
+
+   if (add_td)
+      fprintf(f, "</td>");
+}
+
+static void cover_print_bin(FILE *f, cover_pair_t *pair, uint32_t flag,
+                            cov_pair_kind_t pkind, int cols, ...)
+{
+   va_list argp;
+   va_start(argp, cols);
+
+   if (pair->flags & flag) {
+      fprintf(f, "<tr><td><b>Bin</b></td>");
+
+      for (int i = 0; i < cols; i++) {
+         const char *val = va_arg(argp, const char *);
+         fprintf(f, "<td>%s</td>", val);
+      }
+
+      if (pkind == PAIR_UNCOVERED)
+         cover_print_get_exclude_button(f, pair->tag, flag, true);
+
+      fprintf(f, "</tr>");
+   }
+}
+
+static void cover_print_bin_header(FILE *f, cov_pair_kind_t pkind, int cols, ...)
+{
+   va_list argp;
+   va_start(argp, cols);
+
+   fprintf(f, "<tr style=\"background-color:#999999;\">");
+   fprintf(f, "<th style=\"width:40px;\"></th>");
+
+   for (int i = 0; i < cols; i++) {
+      const char *val = va_arg(argp, const char *);
+      fprintf(f, "<th style=\"width:100px;\">%s</th>", val);
+   }
+
+   if (pkind == PAIR_UNCOVERED)
+         fprintf(f, "<th style=\"width:150px;\">Exclude Command</th>");
+
+   fprintf(f, "</tr>");
+}
+
+static void cover_print_bins(FILE *f, cover_pair_t *pair, cov_pair_kind_t pkind)
 {
    loc_t loc = pair->tag->loc;
+   char t_str[6] = {0};
```

Comment:
These two don't actually need to be arrays as you only write constant strings into them: just make them `const char *`.